clang/include/clang/Sema/Sema.h

```diff
@@ -1759,6 +1759,33 @@ class Sema final {
     }
   };
 
+  /// Bitmask to contain the list of reasons a single diagnostic should be
+  /// emitted, based on its language.  This permits multiple offload systems
+  /// to coexist in the same translation unit.
+  enum DeviceDiagnosticReason {
```

Comment:
Any reason not to make this an enum class and drop the prefix from the enumerators? People won't likely need to spell these enumerators out all that often. We should also consider using this interface: https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/ADT/BitmaskEnum.h so that it's easier to use these enumerations as a type-safe bitmask.

Reply:
I chose old-style-enums so we could do a bitmask.  I was unaware of BitmaskEnum though, so that seems like a good improvement.

Reply:
Done.